### PR TITLE
Save comment

### PR DESCRIPTION
**Pull request by gabelula**
- It changes the response to sponged  endpoint  PUT /v1/item. Now it returns 200 and the item with the item_id instead of StatusNoContent. 

- It changes the Sponged endpoints from /1.0/ to /v1/.

To test it, run Sponged and Corald locally. 

curl -X PUT -H "Content-Type: application/json" -d '{ "type": "comments", "version": 1, "data": { "_id": "57c489028835dfd623fb9369", "assetID": "3440983", "commentBody": "Testing Body."}}' "127.0.0.1:16180/v1/item"

It should return a JSON with item_id key.

### CODE REVIEW

**Comment by dwhitena** [comment]
I think we need to pull in master here, resolve conflicts, and then have a review.  I want to make sure that the `itMap` will always have the right ID depending on how the item is upserted (inserted or updated).

**Comment by dwhitena** [comment]
I think we may need to make a slight modification to sponge here to make sure a new UUID generated for an item is returned correctly to the `itMap`, but we can review and discuss.

**Comment by gabelula** [comment]
@dwhitena what are you thinking/meaning about  "a new UUID generated for an item is returned correctly" ?  Right now it is being created by uuid if it does not exist.

File: cmd/sponged/handlers/item.go [hunk 1]
```diff
@@ -101,7 +101,8 @@ func (itemHandle) Upsert(c *app.Context) error {
 		return err
 	}
 
-	c.Respond(nil, http.StatusNoContent)
+	c.Respond(itMap, http.StatusOK)
```
Comment (ardan-bkennedy) [inline cmd/sponged/handlers/item.go]: `it` and `itMap` are not good variable names.  Try  `itm` and `itmMap`

Comment (ardan-bkennedy) [inline cmd/sponged/handlers/item.go]: When I say the variable I say  it  or  it map.  I think the context is being lost.

Comment (dwhitena) [inline cmd/sponged/handlers/item.go]: Good point.

**Comment by dwhitena** [comment]
@gabelula Nevermind my comment about the UUID.  I think that part is fine.

**Comment by dwhitena** [comment]
@gabelula Note as well that changing the code returned (because we are not longer returning NoContent) will break the tests.  So, we should update the tests for `sponged` as well.

**Comment by ardan-bkennedy** [comment]
I am worried about these changes to the API. Everything here is not only internally breaking changes but it breaks the front end. Specially if we do this for Xenia. We broke Xenia already with changing /v1/ in the URL, we have not broke how these API's are handled.

**Comment by gabelula** [comment]
Yes. The tests are changed now. Sorry for the other PR that didn't. 

About front end (FE), yes. This should be quite a easy change in FE and I'm going to talk with them today.

**Comment by dwhitena** [comment]
@ardan-bkennedy noted and I agree.  I think the main issue with the change of the return code, was that the FE wanted/needed the item ID returned on insert.  However, I agree that we should take extra care in changes things like this, because this will then create inconsistencies with how other APIs (e.g., xeniad) are currently responding.

File: cmd/sponged/handlers/item.go (hunk 1 above)
Comment (jde) [inline cmd/sponged/handlers/item.go]: Disemvoweling is the best way to change letters while maintaining maximum readability.

File: cmd/corald/routes/routes.go [hunk 2]
```diff
@@ -65,6 +65,9 @@ func routes(a *app.App) {
 				return "/v1/exec/" + c.Params["query_set"]
 			}))
 
-	a.Handle("POST", "/v1/item", fixtures.Handler("items/itemid", http.StatusCreated))
-	a.Handle("PUT", "/v1/item", handlers.Proxy(spongedURL, nil))
+	a.Handle("PUT", "/v1/item",
```
Comment (jde) [inline cmd/corald/routes/routes.go]: I propose we accept both PUT and POST for the upsert endpoint. This follows the clear web semantics around CRUD: POST for Create, PUT for Update. "Upsert" does both so we should accept both.

**Comment by jde** [comment]
The xenia proxy endpoint is a bit confusing:

https://github.com/coralproject/shelf/blob/0c2379e76f5315d040be7a1587ad6d9400500a3d/cmd/corald/routes/routes.go#L62

```/v1/item/:view_name/:item_key/:query_set```

Having "item" followed by a view_name is going to lead to confusion.  I recommend we either:

```/v1/view/:view_name/:item_key/query/:query_set```

Which would read, "start with this view on this item_id and run this query_set", or

```/v1/query/:query_set/view/:view_name/:item_key```

Which reads "run this query_set against this view on this item_id"

**Pull request merged**